After upgrading to 1.5.0 I get 403 on from image retrieval

<!--
Before filing an issue, please reproduce with the latest version of Jib.

If you are encountering errors pulling or pushing to remote registries,
please check our Frequently Asked Questions before filing an issue:

    https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#registry-errors
-->

**Environment**:

- *Jib version:* 1.5.0
- *Build tool:* <!-- Maven/Gradle, including version --> mvn 3.6.1
- *OS:* Windows 10

**Description of the issue**:
After upgrading from 1.4.0 to 1.5.0 I get the following error when trying to create a docker build,

```
[INFO] Containerizing application to Docker daemon as microsvc-operational-information...
[INFO] Executing tasks:
[INFO] [=========                     ] 30,0% complete
[INFO] > pulling base image manifest
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.477 s
[INFO] Finished at: 2019-08-19T10:48:23+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.5.0:dockerBuild (default-cli) on project operational-information-ms: com.google.cloud.tools.jib.api.RegistryUnauthorizedException: Unauthorized for mcr.microsoft.com/dotnet/core/aspnet: 403 Forbidden
[ERROR] ERROR 403: Time-Limited URL validation failed
[ERROR] -> [Help 1]
[ERROR]
```

Downgrading to 1.4.0 again runs as before.

**Expected behavior**:

Build executes normally.

**Steps to reproduce**:
<!-- Please provide a minimal and precise series of steps -->

  1. Run `mvn jib:dockerBuild ` with image mcr.microsoft.com/dotnet/core/aspnet:2.1-stretch-slim
  2. Build fails.
  3. 

**`jib-maven-plugin` Configuration**: <!-- Delete this section if not used -->
```xml
            <plugin>
                <!-- https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin -->
                <groupId>com.google.cloud.tools</groupId>
                <artifactId>jib-maven-plugin</artifactId>
                <version>1.5.0</version>
                <configuration>
                    <container>
                        <!-- https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#container-object -->
                        <entrypoint>
                            <command>dotnet</command>
                            <arg>Operational_Information.dll</arg>
                        </entrypoint>
                        <ports>
                            <port>80</port>
                            <port>443</port>
                        </ports>
			<user>1001</user>
                        <workingDirectory>/netcoreapp2.1/publish</workingDirectory>
			<creationTime>USE_CURRENT_TIMESTAMP</creationTime>
                    </container>
                    <from>
                        <!-- https://hub.docker.com/_/microsoft-dotnet-core -->
                        <image>mcr.microsoft.com/dotnet/core/aspnet:2.1-stretch-slim</image>
                    </from>
                    <to>
                        <image>${docker.image.local.name}</image>
                    </to>
                    <extraDirectories>
                        <paths>
                            <path>Operational_Information/bin/Release/</path>
                        </paths>
                    </extraDirectories>
                    <dockerClient>
                        <environment>
                            <key3>value3</key3>
                            <key4>value4</key4>
                        </environment>
                    </dockerClient>
                    <!-- for now, later require cert enhanced JVM -->
                    <allowInsecureRegistries>true</allowInsecureRegistries>
                </configuration>
            </plugin>
```

**Log output**: <!-- If applicable, provide relevant log output -->

**Additional Information**: <!-- Any additional information that may be helpful -->


<!-- Thanks for contributing! -->

--- a/jib-core/CHANGELOG.md
+++ b/jib-core/CHANGELOG.md
@@ -9,6 +9,8 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 
+- Fixed an issue interacting with certain registries due to changes to URL handling in the underlying Apache HttpClient library. ([#1924](https://github.com/GoogleContainerTools/jib/issues/1924))
+
 ## 0.10.1
 
 ### Added

--- a/jib-core/build.gradle
+++ b/jib-core/build.gradle
@@ -43,6 +43,11 @@ dependencies {
   // for best compatibility.
   implementation 'com.google.http-client:google-http-client:1.31.0'
   implementation 'com.google.http-client:google-http-client-apache-v2:1.31.0'
+  // TODO: remove once https://github.com/googleapis/google-http-java-client/issues/795 is fixed and released.
+  // Forcing to downgrade this to 4.5.6 fixes https://github.com/GoogleContainerTools/jib/issues/1914
+  implementation('org.apache.httpcomponents:httpclient:4.5.6') {
+    force = true
+  }
   implementation 'com.google.auth:google-auth-library-oauth2-http:0.16.2'
   implementation 'com.google.guava:guava:28.0-jre'
 

--- a/jib-core/build.gradle
+++ b/jib-core/build.gradle
@@ -41,7 +41,7 @@ dependencies {
   // For Google libraries, check <http-client-bom.version>, <google.auth.version>, <guava.version>,
   // ... in https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/pom.xml
   // for best compatibility.
-  /
+  //
   // TODO: remove once https://github.com/googleapis/google-http-java-client/issues/795 is fixed and released.
   // Forcing to downgrade this to 4.5.6 fixes https://github.com/GoogleContainerTools/jib/issues/1914
   // However, #795 and upgrading httpclient alone may not fix #1914. We may need to explicitly disable URI

--- a/jib-core/build.gradle
+++ b/jib-core/build.gradle
@@ -41,16 +41,21 @@ dependencies {
   // For Google libraries, check <http-client-bom.version>, <google.auth.version>, <guava.version>,
   // ... in https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/pom.xml
   // for best compatibility.
-  implementation 'com.google.http-client:google-http-client:1.31.0'
-  implementation 'com.google.http-client:google-http-client-apache-v2:1.31.0'
+  /
   // TODO: remove once https://github.com/googleapis/google-http-java-client/issues/795 is fixed and released.
   // Forcing to downgrade this to 4.5.6 fixes https://github.com/GoogleContainerTools/jib/issues/1914
   // However, #795 and upgrading httpclient alone may not fix #1914. We may need to explicitly disable URI
   // normalization as discussed in #795.
-  implementation('org.apache.httpcomponents:httpclient:4.5.6') {
-    force = true
+  implementation('com.google.http-client:google-http-client:1.31.0') {
+    exclude group: "org.apache.httpcomponents", module: "httpclient"
   }
-  implementation 'com.google.auth:google-auth-library-oauth2-http:0.16.2'
+  implementation('com.google.http-client:google-http-client-apache-v2:1.31.0') {
+    exclude group: "org.apache.httpcomponents", module: "httpclient"
+  }
+  implementation('com.google.auth:google-auth-library-oauth2-http:0.16.2') {
+    exclude group: "org.apache.httpcomponents", module: "httpclient"
+  }
+  implementation 'org.apache.httpcomponents:httpclient:4.5.6'
   implementation 'com.google.guava:guava:28.0-jre'
 
   implementation 'org.apache.commons:commons-compress:1.18'

--- a/jib-core/build.gradle
+++ b/jib-core/build.gradle
@@ -45,6 +45,8 @@ dependencies {
   implementation 'com.google.http-client:google-http-client-apache-v2:1.31.0'
   // TODO: remove once https://github.com/googleapis/google-http-java-client/issues/795 is fixed and released.
   // Forcing to downgrade this to 4.5.6 fixes https://github.com/GoogleContainerTools/jib/issues/1914
+  // However, #795 and upgrading httpclient alone may not fix #1914. We may need to explicitly disable URI
+  // normalization as discussed in #795.
   implementation('org.apache.httpcomponents:httpclient:4.5.6') {
     force = true
   }

--- a/jib-gradle-plugin/CHANGELOG.md
+++ b/jib-gradle-plugin/CHANGELOG.md
@@ -9,6 +9,8 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 
+- Fixed an issue interacting with certain registries due to changes to URL handling in the underlying Apache HttpClient library. ([#1924](https://github.com/GoogleContainerTools/jib/issues/1924))
+
 ## 1.5.0
 
 ### Added

--- a/jib-gradle-plugin/build.gradle
+++ b/jib-gradle-plugin/build.gradle
@@ -61,12 +61,16 @@ configurations {
 
 dependencies {
   // These are copied over from jib-core and are necessary for the jib-core sourcesets.
-  compile 'com.google.http-client:google-http-client:1.31.0'
-  compile 'com.google.http-client:google-http-client-apache-v2:1.31.0'
-  compile('org.apache.httpcomponents:httpclient:4.5.6') {
-    force = true
+  compile('com.google.http-client:google-http-client:1.31.0') {
+    exclude group: "org.apache.httpcomponents", module: "httpclient"
   }
-  compile 'com.google.auth:google-auth-library-oauth2-http:0.16.2'
+  compile('com.google.http-client:google-http-client-apache-v2:1.31.0') {
+    exclude group: "org.apache.httpcomponents", module: "httpclient"
+  }
+  compile('com.google.auth:google-auth-library-oauth2-http:0.16.2') {
+    exclude group: "org.apache.httpcomponents", module: "httpclient"
+  }
+  compile 'org.apache.httpcomponents:httpclient:4.5.6'
   compile 'com.google.guava:guava:28.0-jre'
 
   compile 'org.apache.commons:commons-compress:1.18'

--- a/jib-gradle-plugin/build.gradle
+++ b/jib-gradle-plugin/build.gradle
@@ -63,6 +63,9 @@ dependencies {
   // These are copied over from jib-core and are necessary for the jib-core sourcesets.
   compile 'com.google.http-client:google-http-client:1.31.0'
   compile 'com.google.http-client:google-http-client-apache-v2:1.31.0'
+  compile('org.apache.httpcomponents:httpclient:4.5.6') {
+    force = true
+  }
   compile 'com.google.auth:google-auth-library-oauth2-http:0.16.2'
   compile 'com.google.guava:guava:28.0-jre'
 

--- a/jib-maven-plugin/CHANGELOG.md
+++ b/jib-maven-plugin/CHANGELOG.md
@@ -9,6 +9,8 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 
+- Fixed an issue interacting with certain registries due to changes to URL handling in the underlying Apache HttpClient library. ([#1924](https://github.com/GoogleContainerTools/jib/issues/1924))
+
 ## 1.5.0
 
 ### Added

--- a/jib-maven-plugin/pom.xml
+++ b/jib-maven-plugin/pom.xml
@@ -63,6 +63,12 @@
       <scope>compile</scope>
     </dependency>
     <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpclient</artifactId>
+      <version>4.5.6</version>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
       <groupId>com.google.auth</groupId>
       <artifactId>google-auth-library-oauth2-http</artifactId>
       <version>0.16.2</version>

--- a/jib-plugins-common/build.gradle
+++ b/jib-plugins-common/build.gradle
@@ -31,6 +31,9 @@ dependencies {
   // Make sure these are consistent with jib-maven-plugin.
   compile 'com.google.http-client:google-http-client:1.31.0'
   compile 'com.google.http-client:google-http-client-apache-v2:1.31.0'
+  compile('org.apache.httpcomponents:httpclient:4.5.6') {
+    force = true
+  }
   compile 'com.google.auth:google-auth-library-oauth2-http:0.16.2'
   compile 'com.google.guava:guava:28.0-jre'
 

--- a/jib-plugins-common/build.gradle
+++ b/jib-plugins-common/build.gradle
@@ -29,12 +29,16 @@ sourceSets {
 
 dependencies {
   // Make sure these are consistent with jib-maven-plugin.
-  compile 'com.google.http-client:google-http-client:1.31.0'
-  compile 'com.google.http-client:google-http-client-apache-v2:1.31.0'
-  compile('org.apache.httpcomponents:httpclient:4.5.6') {
-    force = true
+  compile('com.google.http-client:google-http-client:1.31.0') {
+    exclude group: "org.apache.httpcomponents", module: "httpclient"
   }
-  compile 'com.google.auth:google-auth-library-oauth2-http:0.16.2'
+  compile('com.google.http-client:google-http-client-apache-v2:1.31.0') {
+    exclude group: "org.apache.httpcomponents", module: "httpclient"
+  }
+  compile('com.google.auth:google-auth-library-oauth2-http:0.16.2') {
+    exclude group: "org.apache.httpcomponents", module: "httpclient"
+  }
+  compile 'org.apache.httpcomponents:httpclient:4.5.6'
   compile 'com.google.guava:guava:28.0-jre'
 
   compile 'org.apache.commons:commons-compress:1.18'

